docs/book/src/cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go

```diff
@@ -0,0 +1,200 @@
+			createdCronjob := &cronjobv1.CronJob{}
+
+			// We'll need to retry getting this newly created CronJob, given that creation may not immediately happen.
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, cronjobLookupKey, createdCronjob)
+				if err != nil {
+					return false
+				}
+				return true
+			}, timeout, interval).Should(BeTrue())
+			Expect(createdCronjob.Spec.Schedule).Should(Equal("1 * * * *"))
+			/*
+				Now that we've created a CronJob in our test cluster, the next step is to write a test that actually tests our CronJob controller’s behavior.
+				Let’s test the CronJob controller’s logic responsible for updating CronJob.Status.Active with actively running jobs.
+				We’ll verify that when a CronJob has a single active downstream Job, its CronJob.Status.Active field contains a reference to this Job.
+
+				First, we should get the test CronJob we created earlier, and verify that it currently does not have any active jobs.
+			*/
+			By("CronJob status should initially have no active Jobs")
+			Expect(len(createdCronjob.Status.Active)).Should(Equal(0))
+			/*
+				Next, we actually create a stubbed Job that will belong to our CronJob, as well as its downstream template specs.
+				We set the Job's status's "Active" count to 2 to simulate the Job running two pods, which means the Job is actively running.
+
+				We then take the stubbed Job and set its owner reference to point to our test CronJob.
+				This ensures that the test Job belongs to, and is tracked by, our test CronJob.
+				Once that’s done, we create our new Job instance.
+			*/
+			By("Create active Job underneath CronJob")
+			testJob := &batchv1.Job{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      JobName,
+					Namespace: CronjobNamespace,
+				},
+				Spec: batchv1.JobSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							// For simplicity, we only fill out the required fields.
+							Containers: []v1.Container{
+								{
+									Name:  "test-container",
+									Image: "test-image",
+								},
+							},
+							RestartPolicy: v1.RestartPolicyOnFailure,
+						},
+					},
+				},
+				Status: batchv1.JobStatus{
+					Active: 2,
+				},
+			}
+
+			// Note that your CronJob’s GroupVersionKind is required to set up this owner reference.
+			kind := reflect.TypeOf(cronjobv1.CronJob{}).Name()
+			gvk := cronjobv1.GroupVersion.WithKind(kind)
+
+			controllerRef := metav1.NewControllerRef(createdCronjob, gvk)
+			testJob.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
+			Expect(k8sClient.Create(ctx, testJob)).Should(Succeed())
+			/*
+				Adding this Job to our test CronJob should trigger our controller’s reconciler logic.
+				After that, we can write a test that evaluates whether our controller eventually updates our CronJob’s Status field as expected!
+			*/
+			By("Active Jobs should be incremented to one")
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, cronjobLookupKey, createdCronjob)
```

Comment:
IIRC, you can use assertions inside eventually blocks.  Plus, an assertion function may return a second argument as an error.  Returning a non-nil argument will fail the assertion.  Ergo, I'd do something like:

Eventually(func() ([]string, error) {
  if err := k8sClient.Get(ctx, cronJobLookupKey, createdCronJob); err != nil {
    return nil, err
  }
  var names []string
  for _, job := range createdCronJob.Status.Active {
    names = append(names, job.Name)
  }
  return names, nil
}, timeout, interval).Should(ConsistOf(JobName), "should list our active job %s in the active jobs list in status", JobName)